JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1427,4 +1437,82 @@ Descriptives <- function(jaspResults, dataset, options) {
     return(TRUE)
   else
     return(FALSE)
-}
\ No newline at end of file
+}
+
+.XbarchartNoId <- function(jaspResults, dataset, options) {
+
+  ready <- (length(options$variables) > 1)
+  if (!ready)
+    return()
+
+  jaspResults[["XbarPlot"]] <- createJaspPlot(title = "X bar chart", width = 650, aspectRatio = 1)
+  jaspResults[["XbarPlot"]]$dependOn(c("Xbarchart", "variables"))
+  jaspResults[["XbarPlot"]]$position <- 11
+  XbarPlot <- jaspResults[["XbarPlot"]]
+
+  data1= na.omit(as.data.frame(dataset))
+  means <- rowMeans(data1)
+  subgroups <- 1:length(means)
+  data2 <- data.frame(subgroups = subgroups, means = means)
+  sixsigma <- qcc::qcc(data1, type = 'xbar', plot = F)
+  center <- sixsigma$center
+  sd1 <- sixsigma$std.dev
+  UCL <- max(sixsigma$limits)
+  LCL <- min(sixsigma$limits)
+
+  p <- ggplot2::ggplot(data2, ggplot2::aes(x = subgroups, y = means, group = 1))+ ggplot2::geom_line()+
+    ggplot2::ylab("Subgroup mean")+
+    ggplot2::xlab('Subgroup')+
+    ggplot2::theme_classic()+
+    ggplot2::scale_x_continuous(breaks = c(subgroups), limits = c(1,length(subgroups) + 1.5))+
+    ggplot2::geom_point(size = 2)+
+    ggplot2::geom_hline(yintercept = center, color = 'black')+
+    ggplot2::geom_hline(yintercept = c(center - sd1,center + sd1), linetype = "dashed", color = "red")+
+    ggplot2::geom_hline(yintercept = c(UCL,LCL), color = "red")+
+    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,center,label = sprintf("Mean = %g",round(center,3))))+
+    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,UCL,label = sprintf("UCL = %g",round(UCL,3))))+
+    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,LCL,label = sprintf("LCL = %g",round(LCL,3))))+
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y= ggplot2::element_text(size= ggplot2::rel(1.4)))+
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y= ggplot2::element_text(size= ggplot2::rel(1.4)))+
+    ggplot2::scale_y_continuous(limits = c(LCL,UCL), breaks = NULL)
+
+  XbarPlot$plotObject <- p
+}
+
+.Rchart <- function(jaspResults, dataset, options) {
+
+  ready <- (length(options$variables) > 1)
+  if (!ready)
+    return()
+
+  jaspResults[["RPlot"]] <- createJaspPlot(title = "R chart", width = 650, aspectRatio = 1)
+  jaspResults[["RPlot"]]$dependOn(c("Rchart", "variables"))
+  jaspResults[["RPlot"]]$position <- 11
+  RPlot <- jaspResults[["RPlot"]]
+
+  data1= na.omit(as.data.frame(dataset))
+  range= apply(data1, 1,function(x) {max(range(x)) - min(range(x))})
```

Comment:
`range(x)` is essentially returns `c(max(x), min(x))` so you're calling `max(c(max(x), min(x)))`. Alternatively,
```suggestion
  range <- apply(data1, 1,function(x) max(x) - min(x))
```
or
```suggestion
  range <- apply(data1, 1,function(x) diff(range(x)))
```